build_daemon/lib/client.dart

```diff
@@ -17,22 +17,80 @@ import 'data/build_target_request.dart';
 import 'data/serializers.dart';
 import 'data/server_log.dart';
 
-class BuildDaemonClient {
-  IOWebSocketChannel _channel;
-  Serializers _serializers;
+int _existingPort(String workingDirectory) {
+  var portFile = File(portFilePath(workingDirectory));
+  if (!portFile.existsSync()) throw Exception('Unable to read port file.');
```

Comment:
Use a more specific type, create a new type if you need to.

https://api.dartlang.org/stable/2.1.0/dart-core/Exception-class.html

> Creating instances of Exception directly with `new Exception("message")` is discouraged, and only included as a temporary measure during development, until the actual exceptions used by a library are done.

Same thing below.